packages/smooth_app/lib/pages/product/common/product_refresher.dart

```diff
@@ -119,28 +119,29 @@ class ProductRefresher {
     return const _MetaProductRefresher.error(null);
   }
 
-  Future<bool> fetchAndRefresh({
+  Future<Product?> fetchAndRefresh({
     required final BuildContext context,
     required final LocalDatabase localDatabase,
     required final String barcode,
   }) async {
     final AppLocalizations appLocalizations = AppLocalizations.of(context);
-    final bool? savedAndRefreshed = await LoadingDialog.run<bool>(
+    final _MetaProductRefresher? fetchAndRefreshed =
+        await LoadingDialog.run<_MetaProductRefresher>(
       future: _fetchAndRefresh(localDatabase, barcode),
       context: context,
       title: appLocalizations.nutrition_page_update_running,
     );
-    if (savedAndRefreshed == null) {
-      return false;
+    if (fetchAndRefreshed == null) {
+      return null;
     }
-    if (!savedAndRefreshed) {
+    if (fetchAndRefreshed.product == null) {
       await LoadingDialog.error(context: context);
-      return false;
+      return null;
     }
-    return true;
+    return fetchAndRefreshed.product;
   }
 
-  Future<bool> _fetchAndRefresh(
+  Future<_MetaProductRefresher> _fetchAndRefresh(
```

Comment:
`_saveAndRefresh` should call this method instead of having to copy its code there.

Reply:
fetchAndRefresh calls this method instead